packages/nx-monorepo/src/projects/python/nx-monorepo-py.ts

```diff
@@ -0,0 +1,317 @@
+      throw new Error(
+        `${dependent.name} must be a PythonProject with Poetry enabled to add this dependency`
+      );
+    }
+
+    // Add implicit dependency for build order
+    this.addImplicitDependency(dependent, dependee);
+
+    // Add local path dependency
+    dependent.addDependency(
+      `${dependee.name}@{path="${path.relative(
+        dependent.outdir,
+        dependee.outdir
+      )}", develop=true}`
+    );
+  }
+
+  /**
+   * Ensures that all non-root projects have NxProject applied.
+   * @internal
+   */
+  protected _ensureNxProjectGraph(): void {
+    function _ensure(_project: Project) {
+      if (_project.root === _project) return;
+
+      NxProject.ensure(_project);
+
+      _project.subprojects.forEach((p) => {
+        _ensure(p);
+      });
+    }
+
+    this.subprojects.forEach(_ensure);
+  }
+
+  preSynthesize(): void {
+    // Calling before super() to ensure proper pre-synth of NxProject component and its nested components
+    this._ensureNxProjectGraph();
+
+    super.preSynthesize();
+  }
+
+  /**
+   * @inheritDoc
+   */
+  synth() {
+    this.validateSubProjects();
+    this.setupPythonNx();
+    this.installPythonSubprojects();
+
+    super.synth();
+  }
+
+  /**
+   * @inheritdoc
+   *
+   * NOTE: Be sure to ensure the VIRTUAL_ENV is unset during postSynthesize as the individual poetry envs will only be created if a existing VIRTUAL_ENV cannot be found.
+   */
+  postSynthesize(): void {
+    const vEnv = process.env.VIRTUAL_ENV;
+    process.env.VIRTUAL_ENV = "";
+    super.postSynthesize();
+    process.env.VIRTUAL_ENV = vEnv;
+  }
+
+  /**
+   * Configures the python plugin on NX and ensures local nx dependencies are installed.
+   */
+  private setupPythonNx() {
+    this.nx.plugins.push("@nxlv/python");
+    this.installTask?.exec("npm install --save-dev @nxlv/python@^16 nx@^16");
+  }
+
+  /**
+   * Ensures all python subprojects have their install target called after the monorepo install task.
+   */
+  private installPythonSubprojects() {
+    const installProjects = this.subprojects.filter(
+      (project) => isPythonProject(project) && this.installTask
+    );
+
+    if (this.installTask && installProjects.length > 0) {
+      const nxRunManyInstall = this.composeNxRunManyCommand({
+        target: "install",
+        projects: installProjects.map((project) => project.name),
+        parallel: 1,
+      });
+
+      this.installTask.exec(
+        NodePackageUtils.command.exec(
+          NodePackageManager.NPM,
+          ...nxRunManyInstall
+        )
+      );
+
+      // Ensure that install-py follows dependency order
+      installProjects.forEach((p) => {
+        NxProject.ensure(p).setTarget("install", {
+          dependsOn: ["^install"],
+        });
+      });
+    }
+  }
+
+  /**
+   * Ensures subprojects don't have a default task and that all Python subpackages are configured to use Poetry.
+   */
+  private validateSubProjects() {
```

Comment:
Might want to update the PDKPipeline python project to use poetry :) don't mind if that's a separate change though :)

Reply:
will do.